### DISCUSSION TITLE
[@types/pdfkit] link should accept null values

### DISCUSSION THREAD

**Discussion by yanickrochon** [DefinitelyTyped/DefinitelyTyped#62936] [Issues with a @types package] [answered]
Using PDFKit, the following code produce an error but is totally valid :

```ts
doc.fillColor('red')
    .text(lorem.slice(0, 199), {
      width: 465,
      height: 100,
      columns: 3,
      continued: true
    })
    .fillColor('blue')
    .text(lorem.slice(199, 282), doc.x - 50, doc.y - 10, {
      link: 'http://www.example.com',
      continued: true
    })
    .fillColor('green')
    .text(lorem.slice(182, 400), doc.x + 50, doc.y + 10, {
      link: null,               // <==== warning here
      ellipsis: true,
    });
  ;
```

This is because `link` must be set to `null` in order to stop creating a link in the document. Unfortunately, the provided types ignores this and produce a warning where it shouldn't be.

In short, the following line

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/8ee037e8c7a179ac86928db0bcfafc2a8de5c06d/types/pdfkit/index.d.ts#L187

should be 

```
  link?: string | null | undefined; 
```

[a comment by an automated account is not shown]

**Comment by MysteryBlokHed** [comment, marked as answer]
I just opened #62938

**Comment by BamButz** [comment]
I'm not so sure about this.
If you don't want it to be a link, just don't set the field, so this case should already be covered.

https://github.com/foliojs/pdfkit/blob/3f69586f3cb9d46cb58f255920863bbcbc5f2acd/lib/mixins/text.js#L364

options.link = undefined, (options != null) == false

**Reply by MysteryBlokHed** [reply]
I still feel like it wouldn't be a bad idea, since snippets in the documentation itself use `null`. If `null` were never used in the documentation then I'd agree that the change doesn't help much, but I think it'd be weird for code copied from the documentation to be invalid.

**Reply by yanickrochon** [reply]
Well, since omitting the property does not remove the link when the text element is "continued", then having `null` is a mandatory value to have. In other words, in the example above, without `link: null`, the link continues even if it's not specified. You **have** to have `link: null` to terminate the link from the previous text element.

In short, this is not a "I'm not sure about that" situation, it's part of the actual implementation!

**Reply by BamButz** [reply]
Okay, was not aware about that section in the documentation.
